### PR TITLE
Remove partial field accessors throughout the project

### PR DESCRIPTION
**Pull request by intricate**
Subsumes #192 

All of these field accessors are partial functions that probably aren't necessary to keep around.

### CODE REVIEW

File: src/Crypto/WebAuthn/AttestationStatementFormat/TPM.hs [hunk 1]
```diff
@@ -149,33 +149,53 @@ data TPMSAttest = TPMSAttest
...
-        tpmsrpExponent :: Word32
-      }
-  | TPMSECCParms
-      { tpmsepSymmetric :: Word16,
-        tpmsepScheme :: Word16,
-        tpmsepCurveId :: Cose.CoseCurveECDSA,
-        tpmsepkdf :: Word16
-      }
+  = TPMUPublicParmsRSA TPMSRSAParms
+  | TPMUPublicParmsECC TPMSECCParms
```
Comment (intricate) [inline src/Crypto/WebAuthn/AttestationStatementFormat/TPM.hs:179]: Are you guys okay with this naming convention for the constructors? I'm open to making any changes.

File: src/Crypto/WebAuthn/AttestationStatementFormat/TPM.hs [hunk 2]
```diff
@@ -149,33 +149,53 @@ data TPMSAttest = TPMSAttest
...
+-- | The TPMS_ECC_PARMS structure as specified in
+-- [TPMv2-Part2](https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf)
+-- section 12.2.3.6.
+data TPMSECCParms = TPMSECCParms
+  { tpmsepSymmetric :: Word16,
+    tpmsepScheme :: Word16,
+    tpmsepCurveId :: Cose.CoseCurveECDSA,
+    tpmsepkdf :: Word16
+  }
+  deriving (Eq, Show, Generic, ToJSON)
```
[the remark this answers is not in the document]
Comment (sjoerdvisscher) [reply src/Crypto/WebAuthn/AttestationStatementFormat/TPM.hs:172]: Oh I see we didn't do this either, never mind then :)

**Review by sjoerdvisscher — APPROVED** [review]
I really like how you went the whole way and added new datatypes where the field names actually were useful!

Naming these things is annoying since you have to make up two constructor names for the same thing. Your choice of putting/keeping the algorithm name at the end in the sum types and putting them at the start in the record types makes sense!

File: src/Crypto/WebAuthn/AttestationStatementFormat/TPM.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (intricate) [reply src/Crypto/WebAuthn/AttestationStatementFormat/TPM.hs:172]: > You can add `#page=149` and `#page=150` to link directly to the section.

I had no idea you could link to a specific page in a PDF like that. Thanks!

**Pull request merged**